client.go

```diff
@@ -0,0 +1,155 @@
+/*
+
+Package Client is a Go client for the NHS FHIR API.
+
+For more information about the API visit this link here:
+https://digital.nhs.uk/developer/api-catalogue/personal-demographics-service-fhir#api-description__overview
+
+Usage:
+
+You use this library by creating a new client and calling methods on the client.
+
+Example:
+
+	package main
+
+	import (
+		"github.com/welldigital/nhs-fhir"
+		"fmt"
+	)
+
+	func main() {
+		c, err := client.NewClient()
+		if err != nil {
+			panic(err)
+		}
+
+		patient, err := c.Patient.Get("9000000009")
+	}
+
+*/
+
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+type Client struct {
+	BaseURL   *url.URL
+	UserAgent string
+
+	httpClient *http.Client
+
+	common  service // Reuse a single struct instead of allocating one for each service on the heap.
```

Comment:
is this necessary?

Reply:
I don't think this is a bad thing. The FHIR spec is pretty big so there could be potentially more services in the long term future so setting this way of extension up now isn't going to be do any harm